### DISCUSSION TITLE
Image normailzation

### DISCUSSION THREAD

**Discussion by SebfraOslo** [ultralytics/ultralytics#6382] [Q&A]
Hi,

Are the input images normalized in YOLOv8? If so are the imagenet mean and std used? and where is it defined in the code?

Thank you
Sebastien

**Comment by pderrenger** [comment]
@SebfraOslo hello Sebastien 👋,

Yes, the input images in YOLOv8 are normalized. We use the ImageNet mean and standard deviation values for normalization, which are `[0.485, 0.456, 0.406]` for the means and `[0.229, 0.224, 0.225]` for the standard deviations, respectively.

This normalization is applied as part of the preprocessing pipeline in the code. If you're looking to find or modify these values, you can check the dataset loading and transformation sections of the codebase. Specifically, look for the `transforms` or `datasets` modules where the normalization is typically applied to the images before being fed into the model.

For more detailed information on how to work with datasets and transformations in YOLOv8, you can refer to our documentation on data preparation and customization. 📚

Happy coding!

**Reply by hhaamm** [reply]
I wasn't able to find those values in the code, just the division by 255. E.g.: https://github.com/ultralytics/ultralytics/blob/65ecf1442eef7765732b16fbbaec09fa6936cdb2/ultralytics/engine/predictor.py#L132, https://github.com/ultralytics/ultralytics/blob/65ecf1442eef7765732b16fbbaec09fa6936cdb2/ultralytics/data/loaders.py#L487 and https://github.com/ultralytics/ultralytics/blob/65ecf1442eef7765732b16fbbaec09fa6936cdb2/ultralytics/models/yolo/detect/val.py#L47. I am missing something?

**Reply by pderrenger** [reply]
Hello Sebastien 👋,

Indeed, the default normalization for YOLOv8 images involves only a division by 255 to scale pixel values to [0,1]. The specific ImageNet mean and std normalization isn't applied directly in the core model processing as you observed.

The division by 255 ensures that the network inputs are scaled appropriately, but they are not mean-standardized to ImageNet due to the robustness and generalization capabilities of the YOLO models.

If you need to apply ImageNet normalization explicitly, you can integrate it into the custom dataset loader or preprocessing steps by modifying the image transformation pipeline:

```python
from torchvision import transforms

# Define a transform to normalize the data
transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
])

# Apply this transform to your dataset loader
```

This flexibility allows the model to be more easily adapted to different datasets or specific requirements.

Let me know if you have further questions! Happy modeling! ☺️
